src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1444,12 +1452,21 @@ bool registrationsQuery
      * it has no impact on MongoDB query optimizer */
     queryBuilder.append("$or", entityOr.arr());
     queryBuilder.append(REG_EXPIRATION, BSON("$gt" << (long long) getCurrentTime()));
-    if (attrs.arrSize() > 0) {
+
+    if (attrs.arrSize() > 0)
+    {
         /* If we don't do this checking, the {$in: [] } in the attribute name part will
          * make the query fail*/
         queryBuilder.append(contextRegistrationAttrsNames, BSON("$in" << attrs.arr()));
     }
 
+    //
+    // 'And-in' the service path
+    //
+    const std::string  servicePathString = REG_SERVICE_PATH;
```

Comment:
I understand this is the place were you take servicePath into account for discover (or subscribe availability, which I think also uses registrationQuery()). However, I'm not able to locate the point in which servicePath is taken into account for registrations creation/update... I'm a bit lost in the forest of style-related changes :)

Could you give some hint on where it is?